cypress/e2e/functional/researcher_reports.spec.js

```diff
@@ -0,0 +1,80 @@
+import * as c from '../../support/constants.js'
+import {ResearcherReportsPageElements as researcherReport} from "../../support/elements/researcher_reports_page_elements.js";
+
+context("Researcher reports UI", () => {
+
+  before(function() {
+    cy.visit(c.LEARN_PORTAL_BASE_URL);
+    cy.login(c.ADMIN_USERNAME, c.ADMIN_PASSWORD);
+    cy.visit(c.LEARN_PORTAL_BASE_URL + '/admin');
+    cy.contains("Administration and Reports");
+  });
+
+  function logout() {
+    cy.logout();
+    cy.clearAllCookies();
+  }
+
+  it("Researcher Reports ", () => {
+    cy.log("launch learner reports");
+    cy.contains("Learner Reports").invoke("removeAttr", "target").click();
+    cy.loginPortal(c.ADMIN_USERNAME, c.ADMIN_PASSWORD);
```

Comment:
Why is it necessary to login in the 'before' function *and* here? Shouldn't just one or the other be fine?

Oh wait, I see: it's because you need to login separately to learn-report.portal. ... How about a comment explaining this? And/or perhaps the 'loginPortal' function should be renamed to something like 'loginPortalReport'.

Reply:
As per our current portal setup, we need to login again to the learner reports. Also, the function name 'loginPortal' is a custom command(command.js) and used in other tests too. So, changing that name will affect those tests. So, I left that unchanged.